server.js

```diff
@@ -95,9 +95,12 @@ app.prepare().then(() => {
         )
       })
       res.end()
-    } else if (/^\/doc.*/i.test(pathname)) {
-      // path /doc*/... -> /doc/...
-      let normalized_pathname = pathname.replace(/^\/doc[^?/]*/i, '/doc')
+    } else if (/^\/(doc|docs|documentation)\//i.test(pathname)) {
+      // path /doc*/{item} -> /doc/{item} and trim trailing / if present
+      let normalized_pathname = pathname.replace(
+        /^(\/[^/?]*)(\/.*[^/])?[/?]*/i,
+        '/doc$2'
+      )
```

Comment:
Can you guys help me double check my regexes here?
The first one is simple, `/^\/(doc|docs|documentation)\//`
But the second one, `/^(\/[^/?]*)(\/.*[^/])?[/?]*/`
is a bit confusing. I've tested it extensively by hand but I just don't want to generate bad 301 redirects, which are very hard to undo later. Thanks @iAdramelk @shcheklein

Reply:
see my comment to the this redirect in general. We are solving a non existent problem by supporting this. No reason for this. It should be simplified significantly.

Reply:
> I think it's outdated and we can make it more strict and aligned with what we actually need on the dvc.org . As I mentioned - I can understand docs or documentation being aliases for the doc. But we should stop supporting some random URLs

Yes, I have already simplified it. The first regex is the one that matches the URL path exactly to `/^\/(doc|docs|documentation)\//`. The 2nd one is to normalize it into `/doc/{item}` (without trailing slashes.

But actually I just realized the first one won't always have `/` at the end... And the 2nd one doesn't have to be so complicated since the first one has matched. They can probably be the same one or very similar. I'm looking into this now.